src/lib/y2network/interface_defaults.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yaml"
+
+module Y2Network
+  module InterfaceDefaults
+    # returns default startmode for a new device
+    #
+    # startmode is returned according product, Arch and current device type
+    def new_device_startmode
+      Yast.import "ProductFeatures"
+
+      product_startmode = Yast::ProductFeatures.GetStringFeature(
+        "network",
+        "startmode"
+      )
+
+      startmode = case product_startmode
+      when "ifplugd"
+        if replace_ifplugd?
+          hotplug_usable? ? "hotplug" : "auto"
+        else
+          product_startmode
+        end
+      when "auto"
+        "auto"
+      else
+        hotplug_usable? ? "hotplug" : "auto"
+      end
+
+      startmode
+    end
+
+    # returns a map with device options for newly created item
+    def new_item_default_options
```

Comment:
Hmmm, I am not sure this is good design to have defaults in mixin. I prefer to have separated object that produce new builder as output and can do more reasonable proposal ( like propose device name, be device type specific, check other stuffs and so on). But I do not want to delay this PR more then needed.

Reply:
why not. But currently I was focused on isolating what should be kept and what can be deleted ;-) Especially this initialization to some defaults / proposals is really messy and done at least on three major places so moving the initialization at one place is not that straightforward. I think we will touch this area many times when network-ng will get sharper edges ;-)